### PR TITLE
aws lambda support

### PR DESCRIPTION
**Pull request by davidayalas**
Main changes are an IF to detect if is normal NodeJS or AWS Lambda and the npm package to allow that feature. README update too.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -91,6 +91,22 @@ app.get('/', (req, res) => {
...
+
+if(isInLambda){
+  
+  const awsServerlessExpress = require('aws-serverless-express');
+  const awsServerlessExpressMiddleware = require('aws-serverless-express/middleware');
+
+  app.use(awsServerlessExpressMiddleware.eventContext());
+
+  const server = awsServerlessExpress.createServer(app)
+  exports.handler = (event, context) => awsServerlessExpress.proxy(server, event, context);
```
Comment (vencax) [inline index.js:104]: are you exporting this upwards to other module?

**Comment by vencax** [comment]
Hi, thanks for contrinution. I am not sure whether suggest you to move this pull functionality into separate repo with refactoring the index.js and this repo as a dependency. Or leave it in this even though fraction of users would use it.
Anyone else suggestion?

**Comment by gregorycuellar** [comment]
> Hi, thanks for contrinution. I am not sure whether suggest you to move this pull functionality into separate repo with refactoring the index.js and this repo as a dependency. Or leave it in this even though fraction of users would use it.
> Anyone else suggestion?

Hi,

I suggest to move all except app.listen in a separate file ( ex app.js ) then having 2 separated entry : 

- index.js : for non lambda users => require app.js and then app.listen , it will provide the same behavior as today and users which doesn't use lambda will not be impacted

- lambda.js : for lambda users => require app.js + lambda code

And maybe lambda dependency can be marked as optional.

I can provide a PR if needed.

**Comment by vencax** [comment]
Ok, please do it :) thankyou

Dne st 15. 5. 2019 12:52 uživatel CUELLAR Grégory <notifications@github.com>
napsal:

**Pull request closed without merging**